### PR TITLE
feat: Ensures asymmetric auto-scaling is not defined in the cluster when using the old sharding configuration in `mongodbatlas_advanced_cluster`

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Uses old SDK version when the old schema structure is used for `mongodbatlas_advanced_cluster`

This has been also tested manually creating these 2 clusters:

```hcl
resource "mongodbatlas_advanced_cluster" "test-node-2" {
  project_id   = var.project_id
  name         = "OLD-SCHEMA-CLUSTER"
  cluster_type = "SHARDED"

  replication_specs {
    num_shards = 2 
    region_configs {
      electable_specs {
        instance_size = "M30"
        disk_iops = 3000
        node_count    = 3
      }
      provider_name = "AWS"
      priority      = 7
      region_name   = "EU_WEST_1"
    }
  }
}

resource "mongodbatlas_advanced_cluster" "test-no-node" {
  project_id   = var.project_id
  name         = "NEW-SCHEMA-CLUSTER"
  cluster_type = "SHARDED"

  replication_specs { # first shard
    region_configs {
      electable_specs {
        instance_size = "M10"
        disk_iops = 3000
        node_count    = 3
      }
      provider_name = "AWS"
      priority      = 7
      region_name   = "EU_WEST_1"
    }
  }

  replication_specs { # second shard
    region_configs {
      electable_specs {
        instance_size = "M10"
        disk_iops = 3000
        node_count    = 3
      }
      provider_name = "AWS"
      priority      = 7
      region_name   = "EU_WEST_1"
    }
  }
}
```

and then checking for the auto-scaling mode of each cluster.
For the cluster named `OLD-SCHEMA-CLUSTER` it is `"autoScalingMode" : "CLUSTER",`
For the cluster named `NEW-SCHEMA-CLUSTER` it is `"autoScalingMode" : "SHARD",`

Link to any related issue(s): CLOUDP-285705

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by lantoli** [comment]
@oarbusi cc @EspenAlbert as this is a change to adv_cluster that is not done also in tpf, can you please update the readme file so we don't miss to do this in tpf? https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/internal/service/advancedclustertpf#changes-in-advancedcluster-that-needs-to-be-added-here
thanks

**Comment by oarbusi** [comment]
@lantoli done in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2836/commits/bd76cda01e6a859d46c6a94adec76c15eba14919

File: .changelog/2836.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (AgustinBettati) [inline .changelog/2836.txt:1]: For update operation the following changelog was defined:
```release-note:enhancement
resource/mongodbatlas_advanced_cluster: Adjusts update operation to support cluster tier auto scaling per shard.
```
WDYT if we align this changelog with a similar format?

File: internal/service/advancedcluster/model_advanced_cluster_test.go [hunk 2]
```diff
@@ -463,3 +464,185 @@ func TestCheckRegionConfigsPriorityOrder(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertClusterDescription20241023to20240805(t *testing.T) {
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/model_advanced_cluster_test.go]: Can we have this test defined in a new file `model_sdk_version_conversion_test`?

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 3]
```diff
@@ -483,9 +483,25 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 		return diag.FromErr(err)
 	}
 
-	cluster, _, err := connV2.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
-	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorCreate, err))
+	var clusterName string
+	var clusterID string
+	var err error
+	if isUsingOldAPISchemaStructure(d) {
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster.go]: pass by improvement, I believe this terminology aligns more with our docs (both public and internal)
```suggestion
	if isUsingOldShardingConfiguration(d) {
```

**Review by AgustinBettati — APPROVED** [review]
LGTM! Some minor suggestions

File: .changelog/2836.txt (hunk 1 above)
Comment (oarbusi) [reply .changelog/2836.txt:1]: Done!

File: internal/service/advancedcluster/model_advanced_cluster_test.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/advancedcluster/model_advanced_cluster_test.go]: Good idea, done

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: Done

**Review by cveticm — APPROVED** [review]
lgtm!

**Review by corryroot — APPROVED** [review]
LGTM!

**Comment by oarbusi** [comment]
Test failure is not related to this PR. Merging anyway

**Pull request merged**